libraries/USB/src/USBHIDMouse.h

```diff
@@ -34,26 +34,68 @@
 #define MOUSE_FORWARD   0x10
 #define MOUSE_ALL       0x1F
 
-class USBHIDMouse: public USBHIDDevice {
-private:
-    USBHID hid;
-    uint8_t _buttons;
-    void buttons(uint8_t b);
-    bool write(int8_t x, int8_t y, int8_t vertical, int8_t horizontal);
+#include "./tusb_hid_mouse.h"
+
+enum MousePositioning_t
+{
+  HID_MOUSE_RELATIVE,
+  HID_MOUSE_ABSOLUTE
+};
+
+struct HIDMouseType_t
+{
+  MousePositioning_t positioning;
+  const uint8_t* report_descriptor;
+  size_t descriptor_size;
+  size_t report_size;
+};
+
+extern HIDMouseType_t HIDMouseRel;
+extern HIDMouseType_t HIDMouseAbs;
+
+
+class USBHIDMouseBase: public USBHIDDevice {
 public:
-    USBHIDMouse(void);
+    USBHIDMouseBase(HIDMouseType_t *type);
     void begin(void);
     void end(void);
-
-    void click(uint8_t b = MOUSE_LEFT);
-    void move(int8_t x, int8_t y, int8_t wheel = 0, int8_t pan = 0); 
```

Comment:
if you add the common methods here as virtual, you could cast either mouse as `USBHIDMouseBase` and call it's methods, regardless which type it is. This is just a suggestion. Do not feel obligated to do it :)
```cpp
    virtual void move(int8_t x, int8_t y, int8_t wheel, int8_t pan) = 0;
    virtual void click(uint8_t b) = 0;
    virtual void buttons(uint8_t b) = 0;
```

Reply:
I can't do that with `::move()` as the data types are different for `x` and `y` (relative mouse uses `int8_t` while absolute mouse uses `int16_t`) but I'll add the `::click()` as virtual to the parent class and mark them as override in the child classes